src/config.rs

```diff
@@ -299,8 +305,27 @@ pub fn parse_style_string(style_string: &str) -> Option<ansi_term::Style> {
                         if color_string == "none" && col_fg {
                             None // fg:none yields no style.
                         } else {
+                            // Ensure that get_palette is only called once so only one warning is emitted
+                            // when attempting to retrieve a nonexistent palette
+                            // Assumes that only one palette will be used for every time starship is run
+                            static PALETTE: once_cell::sync::OnceCell<Option<Palette>> =
+                                once_cell::sync::OnceCell::new();
+
```

Comment:
I'd still rather not have a `static` for this, because this might have hard-to-trace effects on tests. Let's just handle the duplicate messages later.